File: output/cpp/tests/simple_test.cpp
#include <string>
#include <unordered_map>
#include <vector>
#include <any>
#include <memory>
#include <functional>



#include <iostream>
#include <string>
#include <cassert>

class SimpleDockerFrameEvent {
public:
    std::string _message;
    std::unordered_map<std::string, std::any> _parameters;

    SimpleDockerFrameEvent(const std::string& message, std::unordered_map<std::string, std::any> params = {})
        : _message(message), _parameters(std::move(params)) {}
};

class SimpleDockerFrameContext {
public:
    SimpleDockerFrameEvent _event;
    std::any _return;
    std::unordered_map<std::string, std::any> _data;

    SimpleDockerFrameContext(SimpleDockerFrameEvent event, std::any default_return = {})
        : _event(std::move(event)), _return(std::move(default_return)) {}
};

class SimpleDockerCompartment {
public:
    std::string state;
    std::unordered_map<std::string, std::any> state_args;
    std::unordered_map<std::string, std::any> state_vars;
    std::unordered_map<std::string, std::any> enter_args;
    std::unordered_map<std::string, std::any> exit_args;
    std::unique_ptr<SimpleDockerFrameEvent> forward_event;
    std::unique_ptr<SimpleDockerCompartment> parent_compartment;

    explicit SimpleDockerCompartment(const std::string& state) : state(state) {}

    std::unique_ptr<SimpleDockerCompartment> clone() const {
        auto c = std::make_unique<SimpleDockerCompartment>(state);
        c->state_args = state_args;
        c->state_vars = state_vars;
        c->enter_args = enter_args;
        c->exit_args = exit_args;
        return c;
    }
};

class SimpleDocker {
private:
    std::vector<std::unique_ptr<SimpleDockerCompartment>> _state_stack;
    std::unique_ptr<SimpleDockerCompartment> __compartment;
    std::unique_ptr<SimpleDockerCompartment> __next_compartment;
    std::vector<SimpleDockerFrameContext> _context_stack;

    void __kernel(SimpleDockerFrameEvent& __e) {
        __router(__e);
        while (__next_compartment) {
            auto next_compartment = std::move(__next_compartment);
            SimpleDockerFrameEvent exit_event("<$");
            __router(exit_event);
            __compartment = std::move(next_compartment);
            if (!__compartment->forward_event) {
                SimpleDockerFrameEvent enter_event("$>");
                __router(enter_event);
            } else {
                auto forward_event = std::move(__compartment->forward_event);
                if (forward_event->_message == "$>") {
                    __router(*forward_event);
                } else {
                    SimpleDockerFrameEvent enter_event("$>");
                    __router(enter_event);
                    __router(*forward_event);
                }
            }
        }
    }

    void __router(SimpleDockerFrameEvent& __e) {
        const auto& state_name = __compartment->state;
        if (state_name == "Start") {
            _state_Start(__e);
        } else if (state_name == "End") {
            _state_End(__e);
        }
    }

    void __transition(std::unique_ptr<SimpleDockerCompartment> next) {
        __next_compartment = std::move(next);
    }

    void _state_Start(SimpleDockerFrameEvent& __e) {
        if (__e._message == "run") {
            printf("SUCCESS: Hello from Docker\n");
            auto __new_compartment = std::make_unique<SimpleDockerCompartment>("End");
            __new_compartment->parent_compartment = __compartment->clone();
            __transition(std::move(__new_compartment));
            return;
        }
    }

    void _state_End(SimpleDockerFrameEvent& __e) {

    }

public:
    SimpleDocker() {
        __compartment = std::make_unique<SimpleDockerCompartment>("Start");
        SimpleDockerFrameEvent __frame_event("$>");
        SimpleDockerFrameContext __ctx(std::move(__frame_event));
        _context_stack.push_back(std::move(__ctx));
        __kernel(_context_stack.back()._event);
        _context_stack.pop_back();
    }

    void run() {
        SimpleDockerFrameEvent __e("run");
        SimpleDockerFrameContext __ctx(std::move(__e));
        _context_stack.push_back(std::move(__ctx));
        __kernel(_context_stack.back()._event);
        _context_stack.pop_back();
    }
};

// Stub functions for placeholder calls
void native() {}
void x() {}

// TAP test harness
int main() {
    printf("TAP version 14\n");
    printf("1..1\n");
    try {
        SimpleDocker s;
        printf("ok 1 - simple_test\n");
    } catch (...) {
        printf("not ok 1 - simple_test\n");
    }
    return 0;
}

File: output/go/tests/simple_test.go

package main

import "fmt"

type SimpleDockerFrameEvent struct {
    _message    string
    _parameters map[string]any
}

type SimpleDockerFrameContext struct {
    _event  SimpleDockerFrameEvent
    _return any
    _data   map[string]any
}

type SimpleDockerCompartment struct {
    state            string
    stateArgs        map[string]any
    stateVars        map[string]any
    enterArgs        map[string]any
    exitArgs         map[string]any
    forwardEvent     *SimpleDockerFrameEvent
    parentCompartment *SimpleDockerCompartment
}

func newSimpleDockerCompartment(state string) *SimpleDockerCompartment {
    return &SimpleDockerCompartment{
        state:    state,
        stateArgs: make(map[string]any),
        stateVars: make(map[string]any),
        enterArgs: make(map[string]any),
        exitArgs:  make(map[string]any),
    }
}

func (c *SimpleDockerCompartment) copy() *SimpleDockerCompartment {
    nc := &SimpleDockerCompartment{
        state: c.state,
        stateArgs: make(map[string]any),
        stateVars: make(map[string]any),
        enterArgs: make(map[string]any),
        exitArgs:  make(map[string]any),
        forwardEvent:     c.forwardEvent,
        parentCompartment: c.parentCompartment,
    }
    for k, v := range c.stateArgs { nc.stateArgs[k] = v }
    for k, v := range c.stateVars { nc.stateVars[k] = v }
    for k, v := range c.enterArgs { nc.enterArgs[k] = v }
    for k, v := range c.exitArgs { nc.exitArgs[k] = v }
    return nc
}

type SimpleDocker struct {
    _state_stack []*SimpleDockerCompartment
    __compartment *SimpleDockerCompartment
    __next_compartment *SimpleDockerCompartment
    _context_stack []SimpleDockerFrameContext
}

func NewSimpleDocker() *SimpleDocker {
    s := &SimpleDocker{}
    s._state_stack = make([]*SimpleDockerCompartment, 0)
    s._context_stack = make([]SimpleDockerFrameContext, 0)
    s.__compartment = newSimpleDockerCompartment("Start")
    s.__next_compartment = nil
    __frame_event := SimpleDockerFrameEvent{_message: "$>", _parameters: nil}
    __ctx := SimpleDockerFrameContext{_event: __frame_event, _data: make(map[string]any)}
    s._context_stack = append(s._context_stack, __ctx)
    s.__kernel(&s._context_stack[len(s._context_stack)-1]._event)
    s._context_stack = s._context_stack[:len(s._context_stack)-1]
    return s
}

func (s *SimpleDocker) __kernel(__e *SimpleDockerFrameEvent) {
    s.__router(__e)
    for s.__next_compartment != nil {
        next_compartment := s.__next_compartment
        s.__next_compartment = nil
        exit_event := &SimpleDockerFrameEvent{_message: "<$", _parameters: s.__compartment.exitArgs}
        s.__router(exit_event)
        s.__compartment = next_compartment
        if s.__compartment.forwardEvent == nil {
            enter_event := &SimpleDockerFrameEvent{_message: "$>", _parameters: s.__compartment.enterArgs}
            s.__router(enter_event)
        } else {
            forward_event := s.__compartment.forwardEvent
            s.__compartment.forwardEvent = nil
            if forward_event._message == "$>" {
                s.__router(forward_event)
            } else {
                enter_event := &SimpleDockerFrameEvent{_message: "$>", _parameters: s.__compartment.enterArgs}
                s.__router(enter_event)
                s.__router(forward_event)
            }
        }
    }
}

func (s *SimpleDocker) __router(__e *SimpleDockerFrameEvent) {
    switch s.__compartment.state {
    case "Start":
        s._state_Start(__e)
    case "End":
        s._state_End(__e)
    }
}

func (s *SimpleDocker) __transition(next *SimpleDockerCompartment) {
    s.__next_compartment = next
}

func (s *SimpleDocker) Run() {
    __e := SimpleDockerFrameEvent{_message: "Run"}
    __ctx := SimpleDockerFrameContext{_event: __e, _data: make(map[string]any)}
    s._context_stack = append(s._context_stack, __ctx)
    s.__kernel(&s._context_stack[len(s._context_stack)-1]._event)
    s._context_stack = s._context_stack[:len(s._context_stack)-1]
}

func (s *SimpleDocker) _state_Start(__e *SimpleDockerFrameEvent) {
    if __e._message == "Run" {
        fmt.Println("SUCCESS: Hello from Docker")
        __compartment := newSimpleDockerCompartment("End")
        __compartment.parentCompartment = s.__compartment.copy()
        s.__transition(__compartment)
        return
    }
}

func (s *SimpleDocker) _state_End(__e *SimpleDockerFrameEvent) {
}

func main() {
	fmt.Println("TAP version 14")
	fmt.Println("1..1")
	_ = NewSimpleDocker()
	fmt.Println("ok 1 - simple_test")
}
